plugin/core/sessions.py

```diff
@@ -125,6 +129,7 @@ def __init__(self,
         self._on_post_initialize = on_post_initialize
         self._on_post_exit = on_post_exit
         self.capabilities = dict()  # type: Dict[str, Any]
+        self.dynamic_capabilities = BidirectionalDictionary()
```

Comment:
Since I'm pretty new to the code, I was confused that `capabilites` actually means `server_capabilities`. Maybe worth renaming? And then also the same for `dynamic_capabilities` probably.

Reply:
And it would be ideal if both are merged into one dict (or a custom form of dict).  :)

Reply:
> And it would be ideal if both are merged into one dict (or a custom form of dict). :)

I agree, this pull request needs more work.